packages/block-editor/src/store/actions.js

```diff
@@ -918,9 +920,18 @@ export const __unstableSplitSelection =
 						  );
 				}
 
-				const length = select.getBlockAttributes( selectionA.clientId )[
-					attributeKeyA
-				].length;
+				const boundAttributes = getBoundAttributesValues(
+					selectionA.clientId,
+					context,
+					registry
+				);
```

Comment:
Shouldn't we disable splitting and merging when an attribute is bound? It doesn't make much sense imo. Binding a paragraph makes it a single entity.

Reply:
Maybe that's another possibility. What should happen when you are editing a bound heading and you press "Enter"?

Reply:
Maybe we can just insert a new block when pressing Enter in a bound block. Do you think that would be better? Any idea how to do that?

Reply:
I think that would be a good first step to disable splitting and merging when in the context of bound attributes. I can't think of a case where someone would intend to create a new block that carries over some part of the edited value to the new block.

In the long run, I would even consider making the more explicit distinction in the user experience that for bound values in RichText you are only editing an external value and shouldn't expect it works the same as with regular attributes.

Reply:
> First, I've created a https://github.com/WordPress/gutenberg/pull/62063 to create a util to transform the block attributes with binding values. This pull request is on top of that one.
> Then, I'm using that helper to use the transformed attributes instead of the fallback: [link](https://github.com/WordPress/gutenberg/pull/61753/files#diff-d1fbd3af72ee0220c5c3ab2952a24d818174aed711b9a50b57c9544218a63988R923-R930).

Assuming you only check whether the attribute has a bound value, you could avoid that logic, and assume that `enter` finishes editing and create a new block with no content inside. That would be similar to the current behavior when handling the value in the readonly mode.

Reply:
I've just tried this approach in [this commit](https://github.com/WordPress/gutenberg/pull/61753/files#diff-d1fbd3af72ee0220c5c3ab2952a24d818174aed711b9a50b57c9544218a63988R875-R884), and it seems to work. That would avoid passing context to the existing `__unstableSplitSelection` parameters.

Is that what you had in mind?

Reply:
Yes, that's exactly what I described.

Reply:
>  What should happen when you are editing a bound heading and you press "Enter"?

Either nothing or create a soft line break (if not disabled for that rich text instance). When you're in a bound attribute it should be completely isolated imo.